helpers/tx.ts

```diff
@@ -34,63 +27,106 @@ export const trackCCTX = (inboundTxHash: string): Promise<void> => {
-              resolve();
-              return;
+          res.forEach((cctxHash: any) => {
+            if (cctxHash) {
+              if (!cctxList[cctxHash]) {
+                cctxList[cctxHash] = [];
+              }
+            }
+            if (!spinnies.spinners[`spinner-${cctxHash}`]) {
+              spinnies.add(`spinner-${cctxHash}`, {
+                text: `${cctxHash}`,
+              });
             }
-          }
+          });
         } catch (error) {}
+      };
+      if (Object.keys(cctxList).length === 0) {
+        spinnies.add(`search`, {
+          text: `Looking for cross-chain transactions (CCTXs) on ZetaChain...\n`,
+        });
+        await fetchCCTX(inboundTxHash);
+      }
+      if (Object.keys(cctxList).length > 0) {
+        spinnies.succeed(`search`, {
+          text: `CCTXs on ZetaChain found.\n`,
+        });
+      }
+      for (const txHash in cctxList) {
+        await fetchCCTX(txHash);
+      }
+      if (Object.keys(cctxList).length > 0) {
+        for (const cctxHash in cctxList) {
+          try {
+            const url = `${API}/zeta-chain/crosschain/cctx/${cctxHash}`;
+            const apiResponse = await axios.get(url);
+            const cctx = apiResponse?.data?.CrossChainTx;
+            const { status, status_message } = cctx.cctx_status;
+            const { sender_chain_id } = cctx.inbound_tx_params;
+            const { receiver_chainId } = cctx.outbound_tx_params[0];
+            const tx = {
+              receiver_chainId,
+              sender_chain_id,
+              status,
+              status_message,
+            };
+            const lastCCTX = cctxList[cctxHash][cctxList[cctxHash].length - 1];
+            const isEmpty = cctxList[cctxHash].length === 0;
+            if (isEmpty || lastCCTX?.status !== status) {
+              cctxList[cctxHash].push(tx);
+              const sender = cctxList[cctxHash]?.[0].sender_chain_id;
+              const receiver = cctxList[cctxHash]?.[0].receiver_chainId;
+              const path = cctxList[cctxHash]
+                .map(
+                  (x) =>
+                    `${clc.bold.underline(x.status)} ${
+                      x.status_message && "(" + x.status_message + ")"
+                    }`
+                )
+                .join(" → ");
+              const text = {
+                text: `${cctxHash}: ${sender} → ${receiver}: ${path}`,
+              };
+              const id = `spinner-${cctxHash}`;
+              switch (status) {
+                case "OutboundMined":
+                case "Reverted":
+                  spinnies.succeed(id, text);
+                  break;
+                case "Aborted":
+                  spinnies.fail(id, text);
+                  break;
+                default:
+                  spinnies.update(id, text);
+                  break;
+              }
+            }
+          } catch (error) {}
+        }
+      }
+      if (
+        Object.keys(cctxList).length > 0 &&
```

Comment:
this should be inside the previus if right? also looks like a very long function, can you refactor it?